### ISSUE TITLE
XLMForSequenceClassification does not work

### ISSUE THREAD

**Issue by Fakekid** [huggingface/transformers#14558]
## Environment info

- `transformers` version: 4.12.5
- Platform:  Linux 3.10.0-957.21.3.el7.x86_64
- Python version: 3.8.5
- PyTorch version (GPU?): 1.8.1-gpu
- Tensorflow version (GPU?): 2.6.0
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help

@LysandreJik 

## Information

Model I am using : XLMForSequenceClassification

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

I use `XLMForSequenceClassification` to train my downstream text classification tasks. 
The loss does not converge and the accuracy of the training set is very similar. The data set used is the English data in `xnli-1.0 (language='en')`. 

When I switch to `BertForSequenceClassification` , it can be trained normally and the acc reached 1.0 on the training set.(The training method is exactly the same.)

The loading method is:
```python
model = XLMForSequenceClassification.from_pretrained('xlm-mlm-tlm-xnli15-1024', num_labels=3)
```

The following prompt appears when loading the model:
```
Some weights of XLMForSequenceClassification were not initialized from the model checkpoint at xlm-mlm-enfr-1024 and are newly initialized: ['sequence_summary.summary.bias','transformer.position_ids','sequence_summary.summary.weight']
```

Is it caused by not including position_ids in the pre-training model?

## To reproduce

1. Load the model.
2. Train the model.
3. Print the acc.

## Expected behavior

The acc after XLMForSequenceClassification training can normally reach the index given in the xlm paper.

Thanks ~~~
